Add support for `SYMPHONY` 11 bit protocol.

crankyoldgit opened the pull request:
* basic `sendSymphony()`/`decodeSymphony()` added.
* Update basic housekeeping to support new protocol.
* Unit tests for new protocol.

Fixes #1057

crankyoldgit committed:
Add support for `SYMPHONY` 11 bit protocol.
* basic `sendSymphony()`/`decodeSymphony()` added.
* Update basic housekeeping to support new protocol.
* Unit tests for new protocol.

Fixes #1057

crankyoldgit committed:
Add a default repeat to SYMPHONY
Per 
https://github.com/crankyoldgit/IRremoteESP8266/issues/1057#issuecomment-596236365

sillyfrog reviewed (approved):
Looks good.

All the work you have done on having generic handlers is paying off! I think there is now more boiler plate code than anything else which is nice :)

crankyoldgit commented:
@sillyfrog thanks! Agreed. Still room for improvement, but it's a lot simpler to add protocols now.

[Edit: Tip to Self: Don't use a mobile phone to respond. Autocorrect sucks!]

crankyoldgit committed:
Tweak timing matching based on feedback.
Ref: 
https://github.com/crankyoldgit/IRremoteESP8266/issues/1057#issuecomment-596543641
